src/Utilities/TargetPlatformSDK.cs

```diff
@@ -94,72 +94,36 @@ public Version MinOSVersion
         /// <summary>
         /// Target platform identifier
         /// </summary>
-        public string TargetPlatformIdentifier
-        {
-            get;
-            private set;
-        }
+        public string TargetPlatformIdentifier { get; }
 
         /// <summary>
         /// Target platform version
         /// </summary>
-        public Version TargetPlatformVersion
-        {
-            get;
-            private set;
-        }
+        public Version TargetPlatformVersion { get; }
 
         /// <summary>
         /// Path to target platform sdk if it exists, it may not if there is no target platform is installed
         /// </summary>
         public string Path
         {
-            get
-            {
-                return _path;
-            }
-
-            set
-            {
-                if (value != null)
-                {
-                    _path = FileUtilities.EnsureTrailingSlash(value);
-                }
-                else
-                {
-                    _path = null;
-                }
-            }
+            get => _path;
+            set => _path = value != null ? FileUtilities.EnsureTrailingSlash(value) : null;
```

Comment:
I don't think it will. `EnsureTrailingSlash` calls `FixFilePath` which does return `null` if the input is null, but then `EnsureTrailingSlash` calls `.Length` on the value of that.

---
In reply to: [197171590](https://github.com/Microsoft/msbuild/pull/3438#discussion_r197171590) [](ancestors = 197171590)

Reply:
Yup, you're right.